compiler/rustc_const_eval/src/interpret/visitor.rs

```diff
@@ -21,7 +21,7 @@ pub trait ValueVisitor<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>>: Sized {
     /// `read_discriminant` can be hooked for better error messages.
     #[inline(always)]
     fn read_discriminant(&mut self, v: &Self::V) -> InterpResult<'tcx, VariantIdx> {
-        Ok(self.ecx().read_discriminant(&v.to_op(self.ecx())?)?)
+        self.ecx().read_discriminant(&v.to_op(self.ecx())?)
```

Comment:
I usually prefer consistently putting in the `?` and Ok-wrapping... but I guess it's futile to fight against clippy lints :/

Reply:
I think in this case, the `())?)?)` is pretty hard to read

Reply:
Yeah, this is an `Ok(foo?)` which peels away one Ok on the inside just to wrap one around again on the outside :upside_down_face:

Reply:
(deleted, misread :sweat_smile: